test/atlas/test_connection.py

```diff
@@ -40,29 +55,47 @@ def _connect(uri):
 
 
 class TestAtlasConnect(unittest.TestCase):
-
-    @classmethod
-    def setUpClass(cls):
-        if not all([_REPL, _SHRD, _FREE]):
-            raise Exception(
-                "Must set ATLAS_REPL/SHRD/FREE env variables to test.")
+    @unittest.skipUnless(HAS_SNI, 'Free tier requires SNI support')
+    def test_free_tier(self):
+        _connect(ATLAS_FREE)
 
     def test_replica_set(self):
-        _connect(_REPL)
+        _connect(ATLAS_REPL)
 
     def test_sharded_cluster(self):
-        _connect(_SHRD)
-
-    def test_free_tier(self):
-        if not HAS_SNI:
-            raise unittest.SkipTest("Free tier requires SNI support.")
-        _connect(_FREE)
+        _connect(ATLAS_SHRD)
 
     def test_tls_11(self):
-        _connect(_TLS11)
+        _connect(ATLAS_TLS11)
 
     def test_tls_12(self):
-        _connect(_TLS12)
+        _connect(ATLAS_TLS12)
+
+    @unittest.skipUnless(HAS_DNS, 'SRV requires dnspython')
+    @unittest.skipUnless(HAS_SNI, 'Free tier requires SNI support')
+    def test_srv_free_tier(self):
+        _connect(ATLAS_SRV_FREE)
+
+    @unittest.skipUnless(HAS_DNS, 'SRV requires dnspython')
+    def test_srv_replica_set(self):
+        _connect(ATLAS_SRV_REPL)
+
+    @unittest.skipUnless(HAS_DNS, 'SRV requires dnspython')
+    def test_srv_sharded_cluster(self):
+        _connect(ATLAS_SRV_SHRD)
+
+    @unittest.skipUnless(HAS_DNS, 'SRV requires dnspython')
+    def test_srv_tls_11(self):
+        _connect(ATLAS_SRV_TLS11)
+
+    @unittest.skipUnless(HAS_DNS, 'SRV requires dnspython')
+    def test_srv_tls_12(self):
+        _connect(ATLAS_SRV_TLS12)
+
+    @unittest.skipUnless(PYMONGO_MUST_HAVE_DNS, 'dnspython is optional')
```

Comment:
Nice! Can you add a comment/docstring detailing what this test achieves? Specifically, we don't end up silently skipping all tests if `dnspython` is not installed (instead we get a failure due to this test).

Reply:
I changed this up a bit. PTAL at the new approach:
> `@unittest.skipUnless(HAS_DNS or MUST_TEST_SRV, 'SRV requires dnspython')`